installer/bundle/bundle_skel_AIX.sh

```diff
@@ -0,0 +1,432 @@
+            if [ "$PKG_NAME" != "*-cimprov" ]; then
+                echo "Removing ${PKG_NAME} ..."
+                pkg_rm ${PKG_NAME}
+            fi
+        done
+
+        # Now just simply pkg_rm scx (and omi if it has no dependencies)
+        pkg_rm scx
+        pkg_rm omi
+    fi
+
+    if [ "$installMode" = "P" ]
+    then
+        echo "Purging all files in cross-platform agent ..."
+        rm -rf /etc/opt/microsoft/*-cimprov /etc/opt/microsoft/scx /opt/microsoft/*-cimprov /opt/microsoft/scx /var/opt/microsoft/*-cimprov /var/opt/microsoft/scx
+        rmdir /etc/opt/microsoft /opt/microsoft /var/opt/microsoft 1>/dev/null 2>/dev/null
+
+        # If OMI is not installed, purge its directories as well.
+        check_if_pkg_is_installed omi
+        if [ $? -ne 0 ]; then
+            rm -rf /etc/opt/omi /opt/omi /var/opt/omi
+        fi
+    fi
+fi
+
+if [ -n "${shouldexit}" ]
+then
+    # when extracting script/tarball don't also install
+    cleanup_and_exit 0
+fi
+
+#
+# Do stuff before extracting the binary here, for example test [ `id -u` -eq 0 ],
+# validate space, platform, uninstall a previous version, backup config data, etc...
+#
+
+#
+# Extract the binary here.
+#
+
+echo "Extracting..."
+
+tail -n +${SCRIPT_LEN_PLUS_ONE} "${SCRIPT}" | gunzip -c | tar xf -
+STATUS=$?
+if [ ${STATUS} -ne 0 ]
+then
+    echo "Failed: could not extract the install bundle."
+    cleanup_and_exit ${STATUS}
+fi
+
+#
+# Do stuff after extracting the binary here, such as actually installing the package.
+#
+
+EXIT_STATUS=0
+SCX_EXIT_STATUS=0
+OMI_EXIT_STATUS=0
+BUNDLE_EXIT_STATUS=0
+
+case "$installMode" in
+    E)
+        # Files are extracted, so just exit
+        cleanup_and_exit 0 "SAVE"
+        ;;
+
+    I)
+        echo "Installing cross-platform agent ..."
+
+        if [ $PROVIDER_ONLY -eq 0 ]; then
+            check_if_pkg_is_installed omi
+            if [ $? -eq 0 ]; then
+                pkg_upd $OMI_PKG omi
+                # It is acceptable that this fails due to the new omi being 
+                # the same version (or less) than the one currently installed.
+                OMI_EXIT_STATUS=0
+            else
+                pkg_add $OMI_PKG omi
+                OMI_EXIT_STATUS=$?
+            fi
+        fi
+
+        pkg_add $OM_PKG scx
+        SCX_EXIT_STATUS=$?
+
+        if [ $PROVIDER_ONLY -eq 0 ]; then
+            # Install bundled providers
+            [ -n "${forceFlag}" ] && FORCE="--force" || FORCE=""
+            for i in *-oss-test.sh; do
+                # If filespec didn't expand, break out of loop
+                [ ! -f $i ] && break
+                ./$i
+                if [ $? -eq 0 ]; then
+                    OSS_BUNDLE=`basename $i -oss-test.sh`
+                    ./${OSS_BUNDLE}-cimprov-*.sh --install $FORCE $restartDependencies
+                    TEMP_STATUS=$?
+                    [ $TEMP_STATUS -ne 0 ] && BUNDLE_EXIT_STATUS="$TEMP_STATUS"
+                fi
+            done
+        fi
```

Comment:
The notion of "mega-bundles" (bundles that install bundles) is a linux-only thing. You should be able to remove 366-380 here, and similar code for all other UNIX-only installers.

I'm being pedantic because now's our chance to get these shell bundles MUCH cleaner than they've ever been before, and thus much easier to maintain.

Make similar changes for HP and Solaris.